src/safeposix/syscalls/fs_calls.rs

```diff
@@ -133,7 +133,60 @@ impl Cage {
         thisfd //open returns the opened file descriptr
     }
 
-    //------------------------------------MKNOD SYSCALL------------------------------------
+    //------------------MKDIR SYSCALL------------------
+
+    pub fn mkdir_syscall(&self, path: &str, mode: u32) -> i32 {
+        //Check that path is not empty
+        if path.len() == 0 {return syscall_error(Errno::ENOENT, "mkdir", "given path was null");}
+
+        let truepath = normpath(convpath(path), self);
+
+        let mut mutmetadata = FS_METADATA.write().unwrap();
+
+        match metawalkandparent(truepath.as_path(), Some(&mutmetadata)) {
+            //If neither the file nor parent exists
+            (None, None) => {
+                syscall_error(Errno::ENOENT, "mkdir", "a directory component in pathname does not exist or is a dangling symbolic link")
+            }
+
+            //If the file doesn't exist but the parent does
+            (None, Some(pardirinode)) => {
+                let filename = truepath.file_name().unwrap().to_str().unwrap().to_string(); //for now we assume this is sane, but maybe this should be checked later
+
+                let effective_mode = S_IFREG as u32 | mode;
```

Comment:
Mkdir does not make a regular file, it makes a directory, replace this with S_IFDIR